javaslang/src/test/java/javaslang/control/TryTest.java

```diff
@@ -247,6 +253,268 @@ public void shouldThrowNullPointerExceptionWhenCallingTryRunRunnable() {
         assertThatThrownBy(() -> Try.runRunnable(null)).isInstanceOf(NullPointerException.class).hasMessage("runnable is null");
     }
 
+    // -- Try.withResources
+
+    @SuppressWarnings("try")/* https://bugs.openjdk.java.net/browse/JDK-8155591 */
+    static class Resource<T> implements AutoCloseable {
+
+        final T value;
+        boolean isClosed = false;
+
+        Resource(T value) {
+            this.value = value;
+        }
+
+        @Override
+        public void close() throws Exception {
+            isClosed = true;
+        }
+    }
+
+    private static <T> Resource<T> resource(T value) {
```

Comment:
It's just my preference: please consider renaming to `Closable` or `TestAutoClosable` or `DummyClosable` etc and provide a static `of` factory, as done in other places.